src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py

```diff
@@ -144,7 +177,8 @@ def on_error(self, peer, exception):
         self.logger.warning(f"EVA transfer error: peer {hexlify(peer.mid)}, exception: {exception}")
 
     def send_remote_select(self, peer, processing_callback=None, force_eva_response=False, **kwargs):
```

Comment:
This function seems to pass its results to either a callback or a feature, depending on the argument passed to `force_eva_response`. I'm not sure why this is, but I think it makes more sense to either stick with callbacks or futures (but not both).

Reply:
You make a very good point here. Indeed, I would prefer `Future`s+`await`s. The only reason callbacks are still there is for compatibility with multiple-UDP packets response for pre-EVA versions. In that case, it is impossible to `await` on `Future`, because we never know if the received packet was the last one.

I'm a bit afraid of moving *all* the transport messages to EVA for 7.10. After 7.10, I will refactor these methods to use `Futures` only :+1:

Reply:
OK, sounds good 👍